### ISSUE TITLE
[A11y] Explore Collections - Grid - Items per page button

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1885]
### Operating System

Windows

### Browser

Edge

### Affected page

Explore Collections

### Admin user?

Yes

### Issue Description

The items per page button has a few problems:

- There's no accessible label that explains this since it's just a button.  The "SHOW:" text next to it isn't actually associated programmatically.
- This is basically a disclosure but isn't coded to act like one.  It actually has some menu coding, but it isn't coded as a full menu widget, either.

### Suggested Fix

This could just be a more traditional "Results per page" set of links with no dropdown element.  I'm guessing we want that for the style, but simplifying the UI might be worth consideration.

If not, this needs to be a fully conformant disclosure or menu.  It's probably easier to make a disclosure, but it's so close to being a conformant menu already.  It needs `aria-haspopup="true"` and potentially `aria-controls="IDREF"` and the JS for some of the missing keyboard functionality.

The label for the button needs to reflect the purpose, not the "current value".  It should be "Show X results per page" or similar.  This label could be hidden, but that might be a problem for other WCAG conformance (for instance for low-vision users who use a very high zoom level and a screen reader).

See https://www.w3.org/TR/wai-aria-practices-1.1/examples/menu-button/menu-button-links.html for an example of a navigation menu button.

### Can anybody QA or does this need an expert?

Needs expert

### QA Steps

Validation of the menu/disclosure needs expertise to ensure all the keyboard controls work, aria is set properly, etc.

**Comment by CGillen** [comment]
Accidentally closed by #1987, that PR does not fix this

**Comment by jechols** [comment]
This issue isn't fixed: the button's label is still "8", and it's still a sort of half-menu, half-disclosure widget, not fully conforming to either pattern.

**Comment by jechols** [comment]
Still not accessible :(

The button's accessible name is still simply "8" (or whatever is selected). The label "show" nearby is insufficient to explain what the button's purpose is, because the button functions as both the selector and the current value, making this very difficult to pull off. I still strongly suggest a change to the design to make it more accessible with less work. If that's absolutely not an option, the button should have some kind of dynamic aria-label such as "Showing 8 results". That's not ideal, but knowing it's a button with an expandable region, it gives as much context as sighted users get.

The button is acting as a navigation disclosure, but the list items are calling themselves a menu. I'd advise going with a disclosure because that makes sense semantically and should be easier to fix. A menu button is fine, too, but likely to be a lot harder to get implemented fully.

- To make it a disclosure:
  - Remove the `role="menu"` from the `<ul>`
  - Remove the `role="menuitem"` from the `<li>` elements
  - Reference: [disclosure example for navigation](https://www.w3.org/TR/wai-aria-practices-1.1/examples/disclosure/disclosure-navigation.html) - this is the most relevant example. The "vanilla" disclosure pattern doesn't require as much work as a navigation disclosure, so pay attention to that. Things like ESC closing the menu are critical because the navigation menu "hovers" over other page content. (this already occurs; I point it out to ensure remediation efforts don't cause it to go away)
- To make it a menubutton:
  - The button needs `aria-haspopup="true"`
  - The keyboard needs to be able to navigate within the menu when pressing the first character of any item. e.g., pressing "2" should bring focus to the number "20".
  - The whole widget needs to be a single tab-stop. Though the sub-elements are focusable, they are not meant to be in the tab sequence. They get focus via the arrow keys.
  - When the user tabs off the menu it must close.
  - The element with the role "menu" (the `<ul>`) needs one of the following:
    - `aria-labelledby` set to a value that refers to the menuitem or button that controls its display.
    - A label provided by `aria-label`.
  - References:
    - [Menu Button pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#menubutton
    - [Menu pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#menu): this has information you have to really dig through. A menu button requires the menu itself implement the right roles/keyboard interaction/etc., but the bits about the menu *bar* have to be ignored.

**Comment by jechols** [comment]
I'm seeing the same problems as before:

- The button still doesn't have any label beyond "8", which is always going to be a hard fail
  - A trivial fix here would be `aria-label="Results per page"`
- Additionally, expected menu widget behaviors aren't fully implemented:
  - Tabbing off the menu is moving to the next link instead of moving out of the menu
  - Hitting a number doesn't move focus to the link starting with said number

The menu widget behaviors might not actually be a hard fail; that stuff is a bit murky. It should be implemented because it's the expected behavior and because it *might* be a hard fail. I'm just not 100% sure about it.

I'm also wondering if this might not be the latest version of the page. It seems like this is almost exactly the same as the last time I tested it....
